pallets/gear-builtin/src/proxy.rs

```diff
@@ -78,14 +78,19 @@ where
     fn handle(
         dispatch: &StoredDispatch,
         context: &mut BuiltinContext,
-    ) -> Result<Payload, BuiltinActorError> {
+    ) -> Result<ActorHandleResult, BuiltinActorError> {
         let request = Request::decode(&mut dispatch.payload_bytes())
             .map_err(|_| BuiltinActorError::DecodingError)?;
 
         let origin = dispatch.source();
 
         let call = Self::cast(request)?;
-        Pallet::<T>::dispatch_call(origin, call, context).map(|_| Default::default())
+
+        Ok(ActorHandleResult {
+            payload: Pallet::<T>::dispatch_call(origin, call, context)
+                .map(|_| Default::default())?,
+            return_value: 0,
```

Comment:
and comment that we don't require value here 
```suggestion
            return_value: dispatch.value(),
```